src/components/PopupWithForm.js

```diff
@@ -1,31 +1,46 @@
-import Popup from '../components/Popup';
+import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit) {
+  constructor(popupSelector, handleSubmit, loadingButtonText) {
     super({ popupSelector });
-    this._handleFormSubmit = handleFormSubmit;
     this._popupForm = this._popupElement.querySelector('.modal__form');
+    this._inputList = this._popupElement.querySelectorAll('.modal__input')
+    this._handleSubmit = handleSubmit;
+    this._formElement = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._formElement.querySelector(".modal__button");
+    this._submitButtonText = this._submitButtonText.textContent;
+    this._loadingText = loadingText;
   }
 
   _getInputValues() {
-    const inputs = this._popupForm.querySelectorAll('.modal__input');
-    const values = {};
-    inputs.forEach(input => {
-      values[input.name] = input.value;
+    const inputValues = {};
+    this._inputList.forEach((input) => {
+      inputValues[input.name] = input.value;
     });
-    return values;
+    return inputValues;
   }
 
-  setEventListeners() {
-    this._popupForm.addEventListener('submit', () => {
-      const inputValues = this._getInputValues();
-      this._handleFormSubmit(inputValues);
-    });
-    super.setEventListeners();
+  _handleSubmit(e) {
+    e.preventDefault();
+    this._handleSubmit(this._getInputValues());
+  }
+
+  open() {
+    this._popupForm.addEventListener('submit', this._handleSubmit);
+    super.open();
   }
 
   close() {
-    this._popupForm.reset();
+    this._popupForm.removeEventListener('submit', this._handleSubmit);
+    this._formElement.reset();
     super.close();
```

Comment:
Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.